components/compliance-service/integration_test/stats_server_read_summary_test.go

```diff
@@ -0,0 +1,323 @@
+
+			//report summary
+			expectedEnvironmentCnt: 1,
+			expectedPlatformCnt:    1,
+			expectedProfileCnt:     1,
+			expectedNodeCnt:        1,
+			expectedStatus:         "passed",
+
+			//nodes summary
+			expectedCompliantCnt:    1,
+			expectedNonCompliantCnt: 0,
+
+			//controls summary
+			expectedPassedCnt: 1,
+		},
+		{
+			description:     "Projects: user has access to unassigned reports",
+			allowedProjects: []string{authzConstants.UnassignedProjectID},
+			query:           octoberTwentyFifthQuery,
+
+			//report summary
+			expectedEnvironmentCnt: 1,
+			expectedPlatformCnt:    1,
+			expectedProfileCnt:     1,
+			expectedNodeCnt:        1,
+			expectedStatus:         "passed",
+
+			//nodes summary
+			expectedCompliantCnt:    1,
+			expectedNonCompliantCnt: 0,
+
+			//controls summary
+			expectedPassedCnt: 1,
+		},
+	}
+
+	for _, test := range successCases {
+		t.Run(test.description, func(t *testing.T) {
+			ctx := contextWithProjects(test.allowedProjects)
+
+			//passing in no type gets us a Summary type that contains a hydrated ReportSummary
+			response, err := statsServer.ReadSummary(ctx, &test.query)
+
+			assert.NoError(t, err)
+			require.NotNil(t, response)
+
+			reportSummary := response.ReportSummary
+
+			//report summary
+			assert.Equal(t, test.expectedEnvironmentCnt, reportSummary.Stats.Environments, "Env count")
+			assert.Equal(t, test.expectedPlatformCnt, reportSummary.Stats.Platforms, "Platforms count")
+			assert.Equal(t, test.expectedProfileCnt, reportSummary.Stats.Profiles, "Profiles count")
+			assert.Equal(t, test.expectedNodeCnt, reportSummary.Stats.Nodes, "Nodes count")
+			assert.Equal(t, test.expectedStatus, reportSummary.Status, "status")
+
+			//passing in "nodes" type gets us a Summary type that contains a hydrated NodeSummary
+			test.query.Type = "nodes"
+			response, err = statsServer.ReadSummary(ctx, &test.query)
+
+			assert.NoError(t, err)
+			require.NotNil(t, response)
+
+			nodeSummary := response.NodeSummary
+
+			//node summary
+			assert.Equal(t, test.expectedCompliantCnt, nodeSummary.Compliant, "Compliant count")
+			assert.Equal(t, test.expectedNonCompliantCnt, nodeSummary.Noncompliant, "NonCompliant count")
+			assert.Equal(t, test.expectedHighRiskCnt, nodeSummary.HighRisk, "HighRisk count")
+			assert.Equal(t, test.expectedLowRiskCnt, nodeSummary.LowRisk, "LowRisk count")
+			assert.Equal(t, test.expectedMediumRiskCnt, nodeSummary.MediumRisk, "MediumRisk count")
+			assert.Equal(t, test.expectedSkippedCnt, nodeSummary.Skipped, "Skipped count")
+
+			//passing in "controls" type gets us a Summary type that contains a hydrated ControlSummary
+			test.query.Type = "controls"
+			response, err = statsServer.ReadSummary(ctx, &test.query)
+
+			assert.NoError(t, err)
+			require.NotNil(t, response)
+
+			controlSummary := response.ControlsSummary
+
+			//control summary
+			assert.Equal(t, test.expectedPassedCnt, controlSummary.Passed, "Passed count")
+		})
+	}
```

Comment:
I'd advocate for splitting these into three tests. That way it'd be a little more obvious where the failure came from if they were to ever fail. I did something like that in the suggestions test:

https://github.com/chef/automate/blob/384a6e3180192aadd658a63c65a65ac481c5f182/components/compliance-service/integration_test/reporting_server_list_suggestions_test.go#L143-L154

Reply:
I agree.. I went back and forth a couple times when I was writing this.   sounds like a good refactor